### ISSUE TITLE
setConfig overrides constructor.

### ISSUE THREAD

**Issue by mathieutu** [SocialiteProviders/Manager#115]
Hi,
as you can see in the following lines, the config is passed twice. Firstly to the constructor of the provider, and then with a `setConfig` method.

https://github.com/SocialiteProviders/Manager/blob/1de3f3d874392da6f1a4c0bf30d843e9cd903ea7/src/SocialiteWasCalled.php#L129-L140

The `setConfig` method is just about filling the config with the passed parameters, and so, totally overriding the previous work from `builProvider` and provider constructor (which could be overridden by the dev)
https://github.com/SocialiteProviders/Manager/blob/1de3f3d874392da6f1a4c0bf30d843e9cd903ea7/src/ConfigTrait.php#L17-L27

It can for example lead to this issue: https://github.com/SocialiteProviders/Manager/pull/110
(actually, I've found that problem, because I had the same)
 
In my opinion we have to purely and simply remove the re-setting of attributes in setConfig.

I can make a quick PR to do that if you want.

Matt'

**Comment by faustbrian** [comment]
Yeah, that one has been bugging me for some time together with with the env config https://github.com/SocialiteProviders/Manager/tree/remove-env-config but I am short on time to spend more hours on it.

A PR to solve this would be very welcome.

**Comment by mathieutu** [comment]
ok, I'll do it as I'll be able to find some time.
I didn't understand what you mean by env config. The link you wrote is broken.

Do you trust in your tests? Should I consider the behaviour ok if the tests pass?

**Comment by faustbrian** [comment]
The link works fine for me. The `env` config is a feature that you can have your config either in the .env file or config/services.php but the issue is that reading it from the .env file makes caching of the impossible so I am going to drop that feature.

AndyWendt wrote the initial tests that are currently used, would probably be a good idea to revise them at some point but for now just make your changes and I will manually test ontop of the tests.

**Comment by xutl** [comment]
I also encountered the same problem.

**Comment by m1guelpf** [comment]
Could anyone PR this?
